pkg/server/sandbox_status.go

```diff
@@ -82,62 +86,70 @@ func (c *criContainerdService) PodSandboxStatus(ctx context.Context, r *runtime.
 
 // TODO (mikebrow): discuss predefining constants structures for some or all of these field names in CRI
 type sandboxInfo struct {
-	Pid         uint32            `json:"pid"`
-	State       string            `json:"state"`
-	SnapshotKey string            `json:"snapshotKey"`
-	Snapshotter string            `json:"snapshotter"`
-	RuntimeSpec *runtimespec.Spec `json:"runtimeSpec"`
+	Pid         uint32                    `json:"pid"`
+	Status      string                    `json:"processStatus"`
+	NetNS       string                    `json:"netNamespace"`
+	NetNSClosed bool                      `json:"netNamespaceClosed"`
+	Image       string                    `json:"image"`
+	SnapshotKey string                    `json:"snapshotKey"`
+	Snapshotter string                    `json:"snapshotter"`
+	Config      *runtime.PodSandboxConfig `json:"config"`
+	RuntimeSpec *runtimespec.Spec         `json:"runtimeSpec"`
 }
 
-// toCRISandboxContainerInfo converts internal container object information to CRI sandbox container status response info map.
-func toCRISandboxContainerInfo(ctx context.Context, container containerd.Container, verbose bool) (map[string]string, error) {
+// toCRISandboxInfo converts internal container object information to CRI sandbox status response info map.
+func toCRISandboxInfo(ctx context.Context, sandbox sandboxstore.Sandbox,
+	pid uint32, processStatus containerd.ProcessStatus, verbose bool) (map[string]string, error) {
 	if !verbose {
 		return nil, nil
 	}
 
-	task, err := container.Task(ctx, nil)
```

Comment:
what was wrong here?

Reply:
I mentioned in the previous comment. It's troublesome to deal with task error (see the logic above).

Since we've got task status in `PodSandboxStatus`, we may want to reuse it.

In the original code here, it's not handling task not found case properly. Note that it's very common that task is deleted but sandbox still exists just not ready.

Reply:
sandbox container task (pause) deleted? But sandbox still exists just not ready.. but we don't need to know that in debug, so we use pod sandbox status?  This is for "extra" verbose debug information, I thought it important to show "raw" task status here in verbose mode vs. cached podsandboxstatus.

Reply:
@mikebrow We want to know whether a sandbox container task state is `stopped` or `deleted`, right?
Actually I don't quite get what you are talking about, sorry. :P